docs/proposals/formatted-output.md

```diff
@@ -0,0 +1,319 @@
+
+- Use xargs utility on Unix
+
+```shell
+oras push $REGISTRY/$REPO:$TAG hello.txt --format '{{.Reference}}' |\
+xargs -I _ oras attach --artifact-type sbom/example _ sbom.spdx --format '{{.Reference}}' |\
+xargs -I _ notation sign _ 
+```
+
+- Use environmental variables on Unix
+
+```shell
+$REFERENCE_A=oras push $REGISTRY/$REPO:$TAG hello.txt --format '{{.Reference}}'
+$REFERENCE_B=oras attach --artifact-type sbom/example $REFERENCE_A sbom.spdx --format '{{.Reference}}' 
+notation sign $REFERENCE_B
+```
+
+- Use [ConverFrom-Json](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.utility/convertfrom-json) on Windows PowerShell
+
+```
+$env:REFERENCE_A='oras push $REGISTRY/$REPO:$TAG hello.txt --format json' --no-tty | ConvertFrom-Json
+$env:REFERENCE_B='oras attach --artifact-type sbom/example $REFERENCE_A.Reference sbom.spdx --format json' --no-tty | ConvertFrom-Json
+notation sign $REFERENCE_B.Reference
+
+```
+
+### CI/CD
+
+Bob is a DevOps engineer. He uses the ORAS GitHub Actions [Setup action](https://github.com/oras-project/setup-oras) to install ORAS in his CI/CD workflow. He wants to chain multiple ORAS commands in a Shell script to perform multiple operations.
+
+For example, pull multiple files (layers) from a repository and filter out the file path of its first layer in the standard output, then pass the pulled first layer to the second command `docker import` for further operation. 
+
+```yaml
+jobs:
+  example-job:
+    steps:
+      - uses: oras-project/setup-oras@v1
+      - run: |
+          oras pull $REGISTRY/$REPO:$TAG --format '{{.first .Files.Path}}' |\
+          docker import _
+```
+
+## Proposal: format output into structured data
+
+1. Use the `--format json` flag to change the default human-readable prettified output to machine-readable raw JSON. Users can still use `--format '{{toPrettyJson .}}'` to get prettified output for some commands.
+2. Use the `--format` with [Go template](https://pkg.go.dev/text/template) to custom the output fields. 
+
+## Desired user experience for proposal 1
+
+For review convenience, this doc shows the output in most of the sample ORAS commands with prettified JSON format.
+
+### oras pull 
+
+Pull an artifact and display its metadata as formatted JSON in standard output. The following fields should be formatted in a JSON output:
+
+- `reference`: full artifact reference by digest, e.g, `$REGISTRY/$REPO:$DIGEST`
+-  `files`: a list of downloaded files
+    -  mediaType: media type of the pulled file (layer) 
+    -  digest: digest of the pulled file (layer) 
+    -  size: file size in bytes
+    -  annotations: annotations of the pulled file (layer)
+    -  path: the absolute file path of the pulled file (layer)
+    -  reference: full reference by digest of the pulled file (layer)
+
+
+Pull a single file and show the manifest of the pulled file as pretty JSON in standard output:
+
+```shell
+oras pull $REGISTRY/$REPO:$TAG --format json {{ toPrettyJson }}
```

Comment:
If we display prettified JSON by default, we don't need this function. 

The most typical use case of `--format json` is to view the schema of output data, prettified output makes more sense since it's easier to read.

Reply:
@sajayantony 
This doc shows the output with prettified JSON format using `--format '{{toPrettyJson .}}'` is just for review convenience. It will be much readable. 

This is one of the function in standard Go template. Users can choose to use this function only when they need to print out prettified JSON.

Reply:
Removed `'{{toPrettyJson .}}'` from all of the samples